haystack/schema.py

```diff
@@ -556,30 +576,261 @@ def append(self, key: str, value: pd.DataFrame):
         else:    
             self.node_results[key] = value
 
-    def calculate_metrics(self) -> Dict[str, float]:
+    def calculate_metrics(
+        self, 
+        simulated_top_k_reader: int = -1,
+        simulated_top_k_retriever: int = -1,
+        doc_relevance_col: str = "gold_id_match"
+    ) -> Dict[str, Dict[str, float]]:
         """
-            First dummy implementation of metrics calcuation just to show the way it's done.
-            TODO: implement retriever and reader specific metrics that must not rely on node names.
+        Calculates proper metrics for each node.
```

Comment:
Can we list the metrics that are calculated by default here?

Reply:
done